### ISSUE TITLE
Bluetooth receiver metadata does not update after directly switching stations

### ISSUE THREAD

**Issue by c00ller** [y20k/transistor#96]
Steps to reproduce:
1) Start playing music from a radio station
2) From the list of radio stations, long press on a different station to begin playing it

Expected behavior: 
Bluetooth receiver's metadata display updates to reflect what is playing on the new station.

Observed behavior:
Bluetooth receiver's metadata display is stuck showing the last information sent by the old station, and never refreshing to show metadata from the new station. The only way to update it is to manually press stop/play. This can also be avoided by stopping the current station before playing a different one.

Despite this, the Transistor notification on the phone always shows the correct, current metadata.

**Comment by y20k** [comment]
Hi, 

someone else did report this behavior via email, too. The problem for me is, that I do not have a Bluetooth receiver. 

But thanks to your description I was able to reproduce the problem using an Android Wear simulator. I guess at least for Wear the problem should be easy to fix. I hope that the root cause is the same. 

I will let you know when I have a solution. Would be great if you could test it.

Best regards

**Comment by y20k** [comment]
The problem should (hopefully) be resolved with yesterday's update. Since I do not have a Bluetooth receiver, it would be great if someone (@c00ller?) could test this. *keepingfingerscrossed*

**Comment by c00ller** [comment]
I just completed a long testing session with the latest update. Some good news, some bad news, and some good news again.

Good news: At the start of my testing, the results were flawless - I kept changing between different stations every few seconds and the correct metadata was updated instantly. I kept doing it and could not break the metadata updating. It was rock solid.

Bad news: That is, until I decided to see how it would handle metadata updating after I started playing music from Google Play Music. After playing some music from GPM, it correctly updated metadata as usual as I skipped tracks in GPM. Nothing unusual here. HOWEVER after I returned to Transistor Radio, Transistor Radio was no longer able to update the metadata. It was very, very inconsistent - there are times when it would work, but >90% of the time, changing a radio station resulted in no updated metadata.

Good news: After much trial and error I found that after this bad behavior was triggered by GPM, Force Closing GPM (or going to Recents and swiping away GPM) will completely fix the state of Transistor Radio again. Changing stations works flawlessly again without any issues (until GPM is opened again).

Basically, GPM seems to "hijack" the metadata updating/handling and will continue to do so until it is force closed in some way. So far I have only tested the interaction with Transistor Radio and Google Play Music (I do not have any other music apps at the moment). I hope this makes sense, and please let me know if you need any clarification or additional testing!

**Comment by y20k** [comment]
Thank you for testing this! It seems that Android Wear and Bluetooth behave pretty similar. Thanks to your detailed description I should be able to reproduce the bug you were describing - using a Wear emulator. I will start looking into this probably next week and I will post the results here.

(PS. I should really get myself a watch ^o^)

**Comment by y20k** [comment]
> 90% of the time, changing a radio station resulted in no updated metadata.

Step by step I am trying to nail the metadata handling. Thank you for pointing out where it still fails!

I think I fixed the problematic behaviour you described. At least (using the Wear emulator) I cannot reproduce it anymore.

**Comment by c00ller** [comment]
Just want to say I haven't forgot about this, I will be able to test it very soon!! Been quite busy lately. Will update this post with results when I have them.

**Comment by y20k** [comment]
No need to hurry. We are going on vacation for a couple of weeks. 🌅🏖

**Comment by c00ller** [comment]
After testing all the scenarios, I am happy to report that this bug has been fixed!

**Comment by y20k** [comment]
Wow. That is good news. Thank you again for doing the testing for me.

--
Vacation is over 😿. I am back.
